### DISCUSSION TITLE
Report printing policy(ascending vs descending)

### DISCUSSION THREAD

**Discussion by Bojun-Seo** [iovisor/bcc#5143] [General]
I thought every report needed to be printed in descending order,
since users are interested in events that happen frequently and consume more time or memory.
And users' interest will be focused on the top of the reports.
However, in some cases, tools like `memleak` may want to print reports in ascending order,
since the reports generated by this tool are usually very long, users can see the last part of the reports by default.

Therefore, I think we need to decide on a report printing policy.
The following are the policy candidates I can imagine.
1) Every tool print reports in descending order
2) Every tool print reports in ascending order

It is obvious that other options can be possible.
Feel free to leave your opinion.
I prefer option `1` by the way.

:::Background of this discussion:::
I recently changed `softirqs` to print report in descending order.
https://github.com/iovisor/bcc/pull/5136
And I found `hardirqs` and `memleak` is printing report in ascending order.
Therefore, I think we need to decide on the policy first.

**Comment by yonghong-song** [comment]
Let us do descending order by default unless there is a strong compelling reason to do otherwise.
